app/views/items/_community_collection_input.html.erb

```
@@ -0,0 +1,30 @@
+<% community_id = local_assigns[:community_id] %>
```

Hmm these should all be required and not optional? Is using `local_assigns` needed here then? Can we just 500/fail hard if they are not supplied? etc?

---

Actually no. When creating a new item, these aren't set so the form needs to handle that (not have a default selected). Also. if a community isn't selected, can't load the list of collections.

---

But your setting them right here? Only way this partial gets rendered I think? Maybe I'm confused?

https://github.com/ualbertalib/jupiter/blob/63f9a173137ebc0d9e717eda354466a127189c4c/app/views/items/_form.html.erb#L23-L26

---

A couple of lines after that, there is an example where they are not set:

https://github.com/ualbertalib/jupiter/blob/63f9a173137ebc0d9e717eda354466a127189c4c/app/views/items/_form.html.erb#L28

---

Ah I see. So why not just pass in the locals here? Instead of these three variable assignments in this file? 
```
<%= render partial: 'community_collection_input', 
                 locals: { f: f, community_id: nil, collection_id: nil, collections: [] } %> ```

---

Done.